test/operators/second_quantization/test_sparse_label_op.py

```diff
@@ -0,0 +1,234 @@
+            test_op = SparseLabelOp(opComplex) * (0.5 + 1j)
+            target_op = SparseLabelOp(
+                {
+                    "+_0 -_1": -0.75 + 1j,
+                    "+_0 -_2": 0.5 + 1j,
+                }
+            )
+
+            self.assertEqual(test_op, target_op)
+
+        with self.subTest("raises TypeError"):
+            with self.assertRaises(TypeError):
+                _ = SparseLabelOp(op1) * "something"
+
+    def test_adjoint(self):
+        """Test adjoint method"""
+        test_op = SparseLabelOp(opComplex).adjoint()
+        target_op = SparseLabelOp(
+            {
+                "+_0 -_1": 0.5 - 1j,
+                "+_0 -_2": 1.0,
+            }
+        )
+        self.assertEqual(test_op, target_op)
+
+    def test_conjugate(self):
+        """Test conjugate method"""
+        test_op = SparseLabelOp(opComplex).conjugate()
+        target_op = SparseLabelOp(
+            {
+                "+_0 -_1": 0.5 - 1j,
+                "+_0 -_2": 1.0,
+            }
+        )
+        self.assertEqual(test_op, target_op)
+
+    def test_transpose(self):
+        """Test transpose method"""
+        test_op = SparseLabelOp(op1).transpose()
+        self.assertEqual(test_op, SparseLabelOp(op1))
+
+    def test_eq(self):
+        """test __eq__ method"""
+        with self.subTest("equal"):
+            test_op = SparseLabelOp(op1) == SparseLabelOp(op1)
+            self.assertTrue(test_op)
+
+        with self.subTest("not equal - keys"):
+            test_op = SparseLabelOp(op1) == SparseLabelOp(
+                {
+                    "+_0 -_1": 0.0,
+                    "+_0 -_3": 1.0,
+                }
+            )
+            self.assertFalse(test_op)
+
+        with self.subTest("not equal - values"):
+            test_op = SparseLabelOp(op1) == SparseLabelOp(op2)
+            self.assertFalse(test_op)
+
+    def test_equiv(self):
+        """test __eq__ method"""
+        with self.subTest("not equivalent - tolerances"):
+            test_op = SparseLabelOp(op1).equiv(
+                SparseLabelOp(
+                    {
+                        "+_0 -_1": 0.000001,
+                        "+_0 -_2": 1.0,
+                    }
+                )
+            )
+
+            self.assertFalse(test_op)
+
+        with self.subTest("not equivalent - keys"):
+            test_op = SparseLabelOp(op1).equiv(
+                SparseLabelOp(
+                    {
+                        "+_0 -_1": 0.0,
+                        "+_0 -_3": 1.0,
+                    }
+                )
+            )
+
+            self.assertFalse(test_op)
+
+        with self.subTest("equivalent"):
+            test_op = SparseLabelOp(op1).equiv(
+                SparseLabelOp(
+                    {
+                        "+_0 -_1": 0.000000001,
+                        "+_0 -_2": 1.0,
+                    }
+                )
+            )
+
+            self.assertTrue(test_op)
```

Comment:
It would be good to do this same test (which passes `equiv`) and ensure it *fails* true equaility  (`__eq__`)